### DISCUSSION TITLE
Git push failed on MacOS due to error env: python: No such file or directory

### DISCUSSION THREAD

**Discussion by anhhchu** [community/community#54475] [Programming Help] [answered]
### Body

I'm really stuck on this issue for a day, and it is driving me crazy. This also never happens before.
When I do `git push origin master` on my local directory. I see below error

```
env: python: No such file or directory
error: failed to push some refs to 'https://github.com/repo_name'
```

I'm on MacOS, my python is alias to python3 

```
$ which python3
/usr/bin/python3

$ which python
python: aliased to /usr/bin/python3

$ python --version
Python 3.9.6
```

I never run into this issue before and as of last week I was still able to do git push normally on this Macbook. This is happening to all repos I have on my local, not just one. 

Really need some help here. Thank you so much

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by airtower-luna** [comment, marked as answer]
The really confusing thing here is why Python is called at all, Git and its basic subcommands are a binary. Do you have any hooks configured in the repository that might actually be Python scripts?

```
$ which python
python: aliased to /usr/bin/python3
```

If that's the case, the fact that `python` is an alias would explain why it can't be found. A hook called from Git won't have access to aliases defined in your shell.

**Reply by anhhchu** [reply]
I don't have any hooks in the git config, all of my hooks configs are sample

```
hooks % ls
applypatch-msg.sample           pre-applypatch.sample           pre-rebase.sample               update.sample
commit-msg.sample               pre-commit.sample               pre-receive.sample
fsmonitor-watchman.sample       pre-merge-commit.sample         prepare-commit-msg.sample
post-update.sample              pre-push.sample                 push-to-checkout.sample
```

**Comment by anhhchu** [comment]
Update on this issue. I recently installed a 3rd party integration with VScode and it installed a git hooks behind the scene. I didn't actually do it, so I wasn't aware. 
To check if this is the case for you, search for `.gitconfig` file in your root directory, and check for hooksPath.  

```
[core]
	# hooksPath = ~/githooks
	python = /usr/bin/python3
```
Hope it helps someone!
